pyartcd/pyartcd/pipelines/brew_scan_osh.py

```diff
@@ -0,0 +1,75 @@
+"""
+[STORES STATE IN REDIS]
+This job scans the candidate tags for a particular version, and triggers scans for builds that are tagged into it.
+"""
+import click
+from typing import Optional
+from pyartcd import exectools, redis
+from pyartcd.runtime import Runtime
+from pyartcd.cli import cli, pass_runtime, click_coroutine
+
+
+class OshScan:
+    def __init__(self, runtime: Runtime, version: str, check_triggered: Optional[bool], email: Optional[str]):
+        self.runtime = runtime
+        self.email = email
+        self.version = version
+        self.major, self.minor = self.version.split(".")
+        self.check_triggered = check_triggered
+
+        self.redis_key = f"appdata:sast-scan:{self.version}"
+
+    async def set_last_event_redis(self, brew_event_id: str):
+        await redis.set_value(self.redis_key, brew_event_id)
+
+    async def get_last_event_redis(self):
+        return await redis.get_value(self.redis_key)
+
+    async def run(self):
+        cmd = [
+            "doozer",
+            "--group",
+            f"openshift-{self.version}",
+            "images:scan-osh",
+        ]
+
+        last_brew_event = await self.get_last_event_redis()
+
+        # Remove trailing /n
+        last_brew_event = last_brew_event.strip()
+
+        if last_brew_event:
+            cmd += [
+                "--since",
+                f"{last_brew_event}"
+            ]
+
+        if self.runtime.dry_run:
+            cmd.append("--dry-run")
+
+        if self.check_triggered:
+            cmd.append("--check-triggered")
+
+        _, brew_event, _ = await exectools.cmd_gather_async(cmd, stderr=True)
+        if not brew_event:
+            self.runtime.logger.error(f"No new builds found for candidate tags in {self.version}")
+            return
```

Comment:
I don't think this exits with an error code? sys.exit(1)?

Reply:
Well we don't want to error out since there might not be builds after an event ID right. Changed to logger.warning if that is better